src/shogun/base/init.cpp

```diff
@@ -37,6 +38,7 @@ namespace shogun
 	Version* sg_version=NULL;
 	CMath* sg_math=NULL;
 	CRandom* sg_rand=NULL;
+	std::unique_ptr<SGLinalg> sg_linalg(nullptr);
```

Comment:
can you please try using shogun's Unique here? if that works then we won't need c++11 guard around all these